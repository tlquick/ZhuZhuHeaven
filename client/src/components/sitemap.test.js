import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Sitemap from "./sitemap";
afterEach(() => {
  cleanup();
});
test("renders sitemap page", () => {
  const { getByText } = render(<Sitemap />);
  const sitemapElement = screen.getByTestId("sitemap-1");
  expect(sitemapElement).toBeInTheDocument();
  expect(getByText("Sitemap")).toBeInTheDocument();
  expect(sitemapElement).toContainHTML("p"); //contains a <p> tag
});
test("matches sitemap snapshot", () => {
  const tree = renderer.create(<Sitemap />).toJSON();
  expect(tree).toMatchSnapshot();
});
