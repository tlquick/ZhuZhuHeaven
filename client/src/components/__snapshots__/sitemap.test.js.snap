// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches sitemap snapshot 1`] = `
<div
  className="main-screen"
  data-testid="sitemap-1"
>
  <h2>
     Sitemap
  </h2>
  <p>
    Lorem ipsum dolor 
  </p>
  <p>
    Qui deleniti velit ea ipsam
  </p>
  <p>
    Sit sequi nostrum qui
  </p>
</div>
`;
